### PR TITLE
Increase deprecation check minimum to iOS 14 and macOS 13

### PR DESCRIPTION
**Pull request by jmagman**
Bump Xcode analyze minimum check version iOS 13 -> iOS 14 and macOS 12.3 -> macOS 13.

- Hit StoreKit2 deprecations. Exclude `in_app_purchase_storekit` to work around StoreKit 1 deprecation warnings.  Added a TODO to remove the exclusion when StoreKit2 is adopted (this should happen in the next few weeks) https://github.com/flutter/flutter/issues/116383.
- Hit https://github.com/flutter/flutter/issues/153587, fixed in https://github.com/flutter/packages/pull/7513
- Hit https://github.com/flutter/flutter/issues/125901 which was `#pragma`'d out in https://github.com/flutter/packages/pull/6221

See related https://github.com/flutter/plugins/pull/5778 and https://github.com/flutter/flutter/issues/102835

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I [linked to at least one issue that this PR fixes] in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style], or this PR is [exempt from CHANGELOG changes].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/blob/master/docs/contributing/Chat.md
[linked to at least one issue that this PR fixes]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#overview
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#version
[following repository CHANGELOG style]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog-style
[exempt from CHANGELOG changes]: https://github.com/flutter/flutter/blob/master/docs/ecosystem/contributing/README.md#changelog
[test-exempt]: https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#tests

### CODE REVIEW

File: script/configs/exclude_xcode_deprecation.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# TODO(louisehsu): Remove deprecation check when StoreKit 2 is adopted. https://github.com/flutter/flutter/issues/116383
```
Comment (jmagman) [inline script/configs/exclude_xcode_deprecation.yaml:1]: @LouiseHsu sorry to assign you a TODO, but this should be good to remove once you have the StoreKit 1 deprecation stuff migrated.
This file and the `--exclude=` part of the target args can be removed.

Comment (LouiseHsu) [reply script/configs/exclude_xcode_deprecation.yaml:1]: np sounds good!

**Comment by stuartmorgan** [comment]
> * Added a TODO to remove the exclusion when StoreKit2 is adopted (this should happen in the next few weeks)

Don't we need to keep StoreKit 1 around indefinitely for iOS 14 and earlier?

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Pull request merged**